sklearn/linear_model/_bayes.py

```diff
@@ -162,7 +174,11 @@ class BayesianRidge(RegressorMixin, LinearModel):
     """
 
     _parameter_constraints: dict = {
-        "n_iter": [Interval(Integral, 1, None, closed="left")],
+        "n_iter": [
+            Interval(Integral, 1, None, closed="left"),
+            Hidden(StrOptions({"deprecated"})),
+        ],
+        "max_iter": [Interval(Integral, 1, None, closed="left"), None],
```

Comment:
`None` is annoying but I don't see how we can do without it.
We will have to deprecate it once `n_iter` is removed.

Reply:
We could use `"warn"` instead to be more explicit, though imo `None` is okay too.

Reply:
`"warn"` will come in 1.5 to detect if a user passed `None` explicitly or not. If we start to add "warn" now, then the only way to not get a warning will be to explicitly set it to an integer. Having `None` now and then `"warn"` will allow to never warn if a user let the default.